odoo_custom_modules/stock_level_alert/models/stock.py

```diff
@@ -0,0 +1,54 @@
+# -*- coding: utf-8 -*-
+from odoo import api, models, _
+from odoo.tools import float_compare
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+    
+    @api.onchange('product_uom_qty', 'product_uom', 'route_id')
+    def _onchange_product_id_check_availability(self):
+        if not self.product_id or not self.product_uom_qty or not self.product_uom:
+            self.product_packaging = False
+            return {}
+        if self.product_id.type == 'product':
+            precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
+            product_qty = self.product_uom._compute_quantity(self.product_uom_qty, self.product_id.uom_id)
+            if float_compare(self.product_id.virtual_available, product_qty, precision_digits=precision) == -1:
+                is_available = self._check_routing()
+                if not is_available:
+                    warning_mess = {
+                        'title': _('Not enough inventory!'),
+                        'message' : _('%s is more than the available quantity') % (self.product_uom_qty)
+                    }
+                    return {'warning': warning_mess}
+        return {}
+        
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+    
+    @api.onchange('product_qty')
+    def _onchange_product_qty(self):
+        #########
+        if not self.product_id or not self.product_qty or not self.product_uom:
+            return
+        if self.product_id.type == 'product':
+            precision = self.env['decimal.precision'].precision_get('Product Unit of Measure')
```

Comment:
@patel-amit this variable isn't being used, could you please double-check and confirm if we don't need it?  If we don't need it, can we please remove it?